AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/Mocks/MockUserDefaults.swift

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+@testable import AWSCognitoAuthPlugin
+
+struct MockUserDefaults: AWSCognitoAuthPluginUserDefaultsBehavior {
+
+    let userDefaults: UserDefaults
```

Comment:
Any reason not to just make this something like a dictionary? No particular reason you should be writing to disk for the unit test.

Reply:
Changed to use NSDictionary inside user defaults mock.